src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py

```diff
@@ -0,0 +1,154 @@
+    session = Session(bind=op.get_bind())
+    conn = session.connection()
+    bind = op.get_bind()
+
+    #############################
+    # legal_basis_for_transfers #
+    #############################
+    statement_transfers = text(
+        """SELECT distinct legal_basis_for_transfers, id
+FROM ctl_systems"""
+    )
+    existing_values_transfers = conn.execute(statement_transfers)
+
+    # drop column since we will repopulating later
+    op.drop_column(
+        "ctl_systems",
+        "legal_basis_for_transfers",
+    )
+
+    # add column back, but make it multi-value
+    op.add_column(
+        "ctl_systems",
+        sa.Column(
+            "legal_basis_for_transfers",
+            type_=sa.ARRAY(sa.String()),
+            server_default="{}",
+            nullable=True,
+        ),
+    )
+
+    # data migration - updates to some values since allowable values enum has changed
+    # (enum is not enforced/specified on the DB, just API-level)
+
+    for row in existing_values_transfers:
+        existing_value = row[0]
+        updated_value = legal_basis_transfers_upgrades.get(
+            existing_value, existing_value
+        )
+        updated_value = [
+            updated_value
+        ]  # wrap in a list because column is now array type
+        bind.execute(
+            text(
+                "UPDATE ctl_systems SET legal_basis_for_transfers = :updated_basis WHERE id = :id"
+            ),
+            {"updated_basis": updated_value, "id": row[1]},
+        )
+
+    #############################
+    # legal_basis_for_profiling #
+    #############################
+    statement_profiling = text(
+        """SELECT distinct legal_basis_for_profiling, id
+FROM ctl_systems"""
+    )
+    existing_values_profiling = conn.execute(statement_profiling)
+
+    op.drop_column(
+        "ctl_systems",
+        "legal_basis_for_profiling",
+    )
+    op.add_column(
+        "ctl_systems",
+        sa.Column(
+            "legal_basis_for_profiling",
+            type_=sa.ARRAY(sa.String()),
+            server_default="{}",
+            nullable=True,
+        ),
+    )
+
+    # data migration - wrap values in a list
+    for row in existing_values_profiling:
+        existing_value = row[0]
+        updated_value = legal_basis_transfers_upgrades.get(
+            existing_value, existing_value
+        )
+        updated_value = [
+            updated_value
+        ]  # wrap in a list because column is now array type
+        bind.execute(
+            text(
+                "UPDATE ctl_systems SET legal_basis_for_profiling = :updated_basis WHERE id = :id"
+            ),
+            {"updated_basis": updated_value, "id": row[1]},
+        )
+
+
+def downgrade():
+    """
+    Downgrade from array fields to single-value.
+
+    We do not try to perform a data migration, since we'd have to
+    drop some values arbitrarily. Better to just null the column out
+    to make it clear that manual intervention is needed.
```

Comment:
what do you mean? can we still migrate what we can?

Reply:
ah multiple values potentially, right

Reply:
maybe a log here

Reply:
added in [1857c7b](https://github.com/ethyca/fides/pull/3977/commits/1857c7b398969554f8d0e0192d674813c7ce0614)

Reply:
thinking more, there's also an argument to be made for keeping the first one? wouldn't most just have one? maybe keep the first and log a list of records where data was lost to reduce downgrade effort. Your call though.

Reply:
yeah, that's a fair argument...i had been weighing it a bit.

one other factor that i'm taking into consideration is that this downgrade is relevant for very few use cases - i don't think we've even officially released a version of the software that has these fields but has them as single-valued, i.e. this downgrade logic really applies to intra-release downgrades only. i think that it basically would have no impact outside of our self-hosted internal instances that have gotten incremental upgrades throughout these past couple of weeks, and i think in the (hopefully unlikely) scenario they need to downgrade, we could live with these fields getting wiped?

so ultimately i felt that it wasn't worth figuring out and testing that logic, but maybe i'm just being a bit lazy 😬

Reply:
no you're right I forgot that it hadn't been released yet, that's fine!